src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,328 @@
+from functools import cache
+from inspect import signature
+from itertools import product
+from typing import Union
+
+import numpy as np
+from sympy import S
+
+from qibo import Circuit, gates, symbols
+from qibo.backends import GlobalBackend
+from qibo.config import raise_error
+from qibo.hamiltonians import SymbolicHamiltonian
+from qibo.transpiler.pipeline import Passes
+from qibo.transpiler.unroller import NativeGates, Unroller
+
+
+@cache
+def _gates(nqubits):
+    """Gates implementing all the GST state preparations.
+
+    Args:
+        nqubits (int): Number of qubits for the circuit.
+    Returns:
+        list(:class:`qibo.gates.abstract.Gate`): list of the gates used to prepare the possible states.
+    """
+
+    return list(
+        product(
+            [(gates.I,), (gates.X,), (gates.H,), (gates.H, gates.S)], repeat=nqubits
+        )
+    )
```

Comment:
@BrunoLiegiBastonLiegi perhaps here we could consider using only native gates for state preparation.
For example just using `GPI2`. However, I'm not sure given that natives gates will be platform dependent...
By running on hardware I could be able to come up with a suitable solution.

Reply:
Indeed, I agree that it would probably be better to use the native gates set of the platform in use, it would actually make a lot of sense to allow the user to define the gates for state preparation, the measurements and the observables, while providing some default configurations. However, I'll leave that to a future PR and I would close this one as soon we verify that it works on real hardware.